resources/prosody-plugins/mod_filter_iq_rayo.lua

```diff
@@ -259,3 +267,27 @@ process_host_module(main_muc_component_host, function(host_module, host)
         end);
     end
 end);
+
+-- when recording participants may enable and backend transcriptions
+-- it is possible that participant is not moderator, but has the features enabled for
+-- transcribing, we need to allow that operation
+module:hook('jitsi-metadata-allow-moderation', function (event)
+    local data, key, occupant, session = event.data, event.key, event.actor, event.session;
+
+    if occupant.role == 'moderator' then
+        return data;
+    end
+
+    if key == 'recording' and data and data.isTranscribingEnabled ~= nil
+        and occupant.role ~= 'moderator'
+        and is_feature_allowed(session.jitsi_meet_context_features, 'transcription')
+        and is_feature_allowed(session.jitsi_meet_context_features, 'recording') then
```

Comment:
Is the check for 'recording' intentional? I thought this was for transcribing, and it's just coincidental that the signaling uses the "recording" key.

Reply:
Yes, it is intentional, this is when you enable recording with backend transcribing via the metadata.
You cannot enable backend transcribing without having recording allowed.